### ISSUE TITLE
FairMQ.SDK: implement a protocol version

### ISSUE THREAD

**Issue by AndreyLebedev** [FairRootGroup/FairMQ#326]
**Is your feature request related to a problem? Please describe.**
Imagine the following example. ODC is built on a node using FairMQ version X. Then ODC starts a topology on a cluster where a FairMQ version Y is used. If there are no changes in the protocol between version X and Y then everything works fine. But if there are incompatible changes then the system stops working correctly.

**Describe the solution you'd like**
Having a protocol version helps to identify a possible incompatibility between two instances communicating via FairMQ.SDK. One can also consider having a schema evolution of the protocol but I think it's overkill.

**Comment by AndreyLebedev** [comment]
Having a handshake with a version number is a good solution. But I would implement a dedicated handshake message for this purpose. Probably you'll want to extend a handshake message in the future with more checks.

**Comment by dennisklein** [comment]
> Having a handshake with a version number is a good solution. But I would implement a dedicated handshake message for this purpose. Probably you'll want to extend a handshake message in the future with more checks.

I agree, looking at the code I realized @rbx was very thourough and already made a `cmd::Cmds` container:

https://github.com/FairRootGroup/FairMQ/blob/868fe02ee9a332b1c7641c1da9bfa0f8e04ac46d/fairmq/sdk/commands/Commands.h#L353-L403

So, we can have a clean, separate handshake type of message and still send it on the same round-trip together with the `cmd::SubscribeToStateChange` message. :tada: :smiley_cat:

**Comment by AndreyLebedev** [comment]
Another option is to return an error / throw an exception from any public API method of `Topology` until the handshake is complete.
